sched/scheduler/jobRunner_test.go

```diff
@@ -40,8 +36,7 @@ func Test_RunTask_PanicWhenWritingStartTaskReturnsFatalError(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	defer mockCtrl.Finish()
 
-	// ensures node does not execute task when a fatal error occurs logging starttask
-	nodeMock := cm.NewMockNode(mockCtrl)
+	worker := worker.NewNoopWorker()
```

Comment:
I dislike removing the mock here.  Which is a critical part of the test.  If we   No way to ensure the task did not get sent to the worker here.  

If the code gets changed and we introduce a bug that sends to the worker or changes the order of operations our unit tests will no longer catch this. 

This is one of the downsides to using mocks instead of fakes